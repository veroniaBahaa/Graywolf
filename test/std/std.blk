rows 3
row -1 11    50 37
row -1 63    50 89
row -1 115    50 141

/* 
   The Tile and Macro Information:
   numtiles 1
   -2 -2   52 232

   nummacros 0

*/
