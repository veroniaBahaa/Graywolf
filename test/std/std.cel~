cell 37 $COUNT_2/$FJK3_4
ECO_added_cell
left -32  right 32  bottom -13  top 13
pin name 00#CD  signal MCLEAR layer 1   -3 13
pin name 00#CP  signal MCLOCK layer 1   1 13
pin name 00#J  signal B0 layer 1   -31 13
pin name 00#K  signal B0 layer 1   -29 13
pin name 00#Q  signal $COUNT_2/$FJK3_4$Q layer 1   27 -13
   equiv name 01#Q layer 1   27 13

cell 38 $COUNT_2/$FJK3_5
left -32  right 32  bottom -13  top 13
pin name 00#CD  signal MCLEAR layer 1   -3 13
pin name 00#CP  signal MCLOCK layer 1   1 13
pin name 00#J  signal IN1 layer 1   -31 13
pin name 00#K  signal IN1 layer 1   -29 13
pin name 00#Q  signal B0 layer 1   27 -13
   equiv name 01#Q layer 1   27 13

cell aslf A!@#$%^&*()_+|-=\[]{};'`:"~,./<>?bn 
/* orient 3 */ 
initially nonfixed 0 from left of block 2
left -2 right 2 bottom -13 top 13 
pin name 00#A signal $COUNT_1/$AND2_1/$ND2_1$Z layer 1 1 13 
pin name 00#Z signal B7 layer 1 -1 -13 
equiv name 01#Z layer 1 -1 13 

cell a $COUNT_1/$AND2_1/$ND2_1 
initially nonfixed 0 from left of block 1
left -3 right 3 bottom -13 top 13 
pin name 101 signal TW_PASS_THRU layer 1 0 13 
equiv name 102 layer 1 0 -13 

cell 14 $COUNT_1/$AND2_1/$ND2_1A 
nomirror 
initially nonfixed 4 from left of block 2
left -3 right 3 bottom -13 top 13 
pin name 00#A signal B6 layer 1 -2 13 
pin name 00#B signal $COUNT_1/$FJK3_1$Q layer 1 0 -13 
equiv name 01#B layer 1 0 13 
pin name 00#Z signal $COUNT_1/$AND2_1/$ND2_1$Z layer 1 2 -13 
equiv name 01#Z layer 1 2 13 



pad 2 name #B0 
corners 4 
3 -28 3 -2 7 -2 7 -28 
orient 0 
restrict side B 
pin name 00##BRISTLE signal B0 layer 1 4 -2 

pad 3 name #B4 
corners 4 
12 -28 12 -2 16 -2 16 -28 
orient 0 
restrict side B 
pin name 00##BRISTLE signal B4 layer 1 13 -2 

pad 4 name #IN1 
corners 4 
21 -28 21 -2 25 -2 25 -28 
orient 0 
restrict side B 
pin name 00##BRISTLE signal IN1 layer 1 22 -2 

pad 5 name #B3 
corners 4 
-28 3 -28 7 -2 7 -2 3 
orient 7 
restrict side L 
pin name 00##BRISTLE signal B3 layer 1 -2 4 

pad 6 name #B7 
corners 4 
-28 12 -28 16 -2 16 -2 12 
orient 7 
restrict side L 
pin name 00##BRISTLE signal B7 layer 1 -2 13 

pad 7 name #MCY 
corners 4 
-28 21 -28 25 -2 25 -2 21 
orient 7 
restrict side L 
pin name 00##BRISTLE signal CONT1 layer 1 -2 22 

pad 8 name #B1 
corners 4 
30 3 30 7 56 7 56 3 
orient 6 
restrict side R 
pin name 00##BRISTLE signal B1 layer 1 30 4 

pad 9 name #B5 
corners 4 
30 12 30 16 56 16 56 12 
orient 6 
restrict side R 
pin name 00##BRISTLE signal B5 layer 1 30 13 

pad 10 name #MCLEAR 
corners 4 
30 21 30 25 56 25 56 21 
orient 6 
restrict side R 
pin name 00##BRISTLE signal MCLEAR layer 1 30 22 

pad 11 name #B2 
corners 4 
3 30 3 56 7 56 7 30 
orient 3 
restrict side T 
pin name 00##BRISTLE signal B2 layer 1 6 30 

pad 12 name #B6 
corners 4 
12 30 12 56 16 56 16 30 
orient 3 
restrict side T 
pin name 00##BRISTLE signal B6 layer 1 15 30 

pad 13 name #MCLOCK 
corners 4 
21 30 21 56 25 56 25 30 
orient 3 
restrict side T 
pin name 00##BRISTLE signal MCLOCK layer 1 24 30 

padgroup padgroup1 nopermute 
#B0 fixed 
#B4 fixed 
#IN1 fixed 
restrict side B 

padgroup padgroup2 nopermute 
#B3 fixed 
#B7 fixed 
#MCY fixed 
restrict side L 

padgroup padgroup3 nopermute 
#B1 fixed 
#B5 fixed 
#MCLEAR fixed 
restrict side R 

padgroup padgroup4 nopermute 
#B2 fixed 
#B6 fixed 
#MCLOCK fixed 
restrict side T 
