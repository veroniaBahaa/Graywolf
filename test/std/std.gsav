total_row_length 144
actual_row_height 26
channel_separation 26
min_length 18
core -2 -2 52 154
grid 0 0
num_macro 0
feed_length 4
spacing 1
numtiles 1
0 -2 -2 52 232 0 1 0 3 26 0 26 18 1 52 0
