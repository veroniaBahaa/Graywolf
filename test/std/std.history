5.00e+00	0.00e+00	3.60e+02	3.56e+03	32	32	-394201182	1.02e+03	2.00000000000000000000000000000000e+00
5.00e+00	0.00e+00	3.60e+02	3.87e+03	32	32	-175941674	1.02e+03	2.00000000000000000000000000000000e+00
5.00e+00	0.00e+00	5.81e+02	1.07e+04	90	90	-249152572	8.10e+03	2.00000000000000000000000000000000e+00
