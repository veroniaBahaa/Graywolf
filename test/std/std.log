####################################################
time 1494792002 Sun May 14 22:00:02 2017:Program initialization completed...
time 1494792002 Sun May 14 22:00:02 2017:Syntax started...
time 1494792002 Sun May 14 22:00:02 2017:Syntax completed...
time 1494792002 Sun May 14 22:00:02 2017:Mincut started...
time 1494792002 Sun May 14 22:00:02 2017:Mincut completed...
time 1494792002 Sun May 14 22:00:02 2017:TimberWolfMC started...
time 1494792008 Sun May 14 22:00:08 2017:TimberWolfMC completed...
time 1494792008 Sun May 14 22:00:08 2017:TimberWolfSC started...
time 1494792009 Sun May 14 22:00:09 2017:TimberWolfSC completed...
####################################################
time 1494792479 Sun May 14 22:07:59 2017:Program initialization completed...
####################################################
time 1494792493 Sun May 14 22:08:13 2017:Program initialization completed...
time 1494792493 Sun May 14 22:08:13 2017:Syntax started...
time 1494792493 Sun May 14 22:08:13 2017:Syntax completed...
time 1494792493 Sun May 14 22:08:13 2017:Mincut started...
time 1494792493 Sun May 14 22:08:13 2017:Mincut completed...
time 1494792493 Sun May 14 22:08:13 2017:TimberWolfMC started...
time 1494792497 Sun May 14 22:08:17 2017:TimberWolfMC completed...
time 1494792497 Sun May 14 22:08:17 2017:TimberWolfSC started...
time 1494792498 Sun May 14 22:08:18 2017:TimberWolfSC completed...
####################################################
time 1494793179 Sun May 14 22:19:39 2017:Program initialization completed...
time 1494793179 Sun May 14 22:19:39 2017:Syntax started...
time 1494793179 Sun May 14 22:19:39 2017:Syntax completed...
time 1494793179 Sun May 14 22:19:39 2017:Mincut started...
time 1494793179 Sun May 14 22:19:39 2017:Mincut completed...
time 1494793179 Sun May 14 22:19:39 2017:TimberWolfMC started...
time 1494793187 Sun May 14 22:19:47 2017:TimberWolfMC completed...
time 1494793187 Sun May 14 22:19:47 2017:TimberWolfSC started...
time 1494793333 Sun May 14 22:22:13 2017:TimberWolfSC completed...
