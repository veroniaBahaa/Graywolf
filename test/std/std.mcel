cluster 1 name core
corners 4 0 0   0 86  86 86   86 0
asplb 0.5 aspub 2.0
class 0 orientations 0 1 2 3 4 5 6 7
softpin name pin1 signal B2
softpin name pin2 signal B5
softpin name pin3 signal B1
softpin name pin4 signal CONT1
softpin name pin5 signal B3
softpin name pin6 signal B4
softpin name pin7 signal B6
softpin name pin8 signal B7
softpin name pin9 signal IN1
softpin name pin10 signal B0
softpin name pin11 signal MCLOCK
softpin name pin12 signal MCLEAR

instance core_L
corners 6 0 0 0 98 49 98 49 49 98 49 98 0
asplb 0.5 aspub 2.0
class 0 orientations 0 1 2 3 4 5 6 7
softpin name pin1 signal B2
softpin name pin2 signal B5
softpin name pin3 signal B1
softpin name pin4 signal CONT1
softpin name pin5 signal B3
softpin name pin6 signal B4
softpin name pin7 signal B6
softpin name pin8 signal B7
softpin name pin9 signal IN1
softpin name pin10 signal B0
softpin name pin11 signal MCLOCK
softpin name pin12 signal MCLEAR

instance core_T
corners 8 43 0 43 43 0 43 0 86 129 86 129 43 86 43 86 0
asplb 0.5 aspub 2.0
class 0 orientations 0 1 2 3 4 5 6 7
softpin name pin1 signal B2
softpin name pin2 signal B5
softpin name pin3 signal B1
softpin name pin4 signal CONT1
softpin name pin5 signal B3
softpin name pin6 signal B4
softpin name pin7 signal B6
softpin name pin8 signal B7
softpin name pin9 signal IN1
softpin name pin10 signal B0
softpin name pin11 signal MCLOCK
softpin name pin12 signal MCLEAR

instance core_L2
corners 6 0 0 0 76 76 76 76 38 114 38 114 0
asplb 0.5 aspub 2.0
class 0 orientations 0 1 2 3 4 5 6 7
softpin name pin1 signal B2
softpin name pin2 signal B5
softpin name pin3 signal B1
softpin name pin4 signal CONT1
softpin name pin5 signal B3
softpin name pin6 signal B4
softpin name pin7 signal B6
softpin name pin8 signal B7
softpin name pin9 signal IN1
softpin name pin10 signal B0
softpin name pin11 signal MCLOCK
softpin name pin12 signal MCLEAR

pad 2 name #B0 
corners 4 
3 -28 3 -2 7 -2 7 -28 
orient 0 
restrict side B 
pin name 00##BRISTLE signal B0 layer 1 4 -2 

pad 3 name #B4 
corners 4 
12 -28 12 -2 16 -2 16 -28 
orient 0 
restrict side B 
pin name 00##BRISTLE signal B4 layer 1 13 -2 

pad 4 name #IN1 
corners 4 
21 -28 21 -2 25 -2 25 -28 
orient 0 
restrict side B 
pin name 00##BRISTLE signal IN1 layer 1 22 -2 

pad 5 name #B3 
corners 4 
-28 3 -28 7 -2 7 -2 3 
orient 7 
restrict side L 
pin name 00##BRISTLE signal B3 layer 1 -2 4 

pad 6 name #B7 
corners 4 
-28 12 -28 16 -2 16 -2 12 
orient 7 
restrict side L 
pin name 00##BRISTLE signal B7 layer 1 -2 13 

pad 7 name #MCY 
corners 4 
-28 21 -28 25 -2 25 -2 21 
orient 7 
restrict side L 
pin name 00##BRISTLE signal CONT1 layer 1 -2 22 

pad 8 name #B1 
corners 4 
30 3 30 7 56 7 56 3 
orient 6 
restrict side R 
pin name 00##BRISTLE signal B1 layer 1 30 4 

pad 9 name #B5 
corners 4 
30 12 30 16 56 16 56 12 
orient 6 
restrict side R 
pin name 00##BRISTLE signal B5 layer 1 30 13 

pad 10 name #MCLEAR 
corners 4 
30 21 30 25 56 25 56 21 
orient 6 
restrict side R 
pin name 00##BRISTLE signal MCLEAR layer 1 30 22 

pad 11 name #B2 
corners 4 
3 30 3 56 7 56 7 30 
orient 3 
restrict side T 
pin name 00##BRISTLE signal B2 layer 1 6 30 

pad 12 name #B6 
corners 4 
12 30 12 56 16 56 16 30 
orient 3 
restrict side T 
pin name 00##BRISTLE signal B6 layer 1 15 30 

pad 13 name #MCLOCK 
corners 4 
21 30 21 56 25 56 25 30 
orient 3 
restrict side T 
pin name 00##BRISTLE signal MCLOCK layer 1 24 30 

padgroup padgroup1 nopermute 
#B0 fixed 
#B4 fixed 
#IN1 fixed 
restrict side B 

padgroup padgroup2 nopermute 
#B3 fixed 
#B7 fixed 
#MCY fixed 
restrict side L 

padgroup padgroup3 nopermute 
#B1 fixed 
#B5 fixed 
#MCLEAR fixed 
restrict side R 

padgroup padgroup4 nopermute 
#B2 fixed 
#B6 fixed 
#MCLOCK fixed 
restrict side T 
