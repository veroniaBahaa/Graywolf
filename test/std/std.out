
TimberWolfSC version:v6.0 date:Mon May 25 21:19:07 EDT 1992
Row-Based Placement and Global Routing Program
Authors: Carl Sechen, Kai-Win Lee, and Bill Swartz,
         Yale University




TimberWolf will perform a global route step
feedThruWidth: 2
track.pitch: 3
route2act was not entered in the .par file
route2act defaulted to track.pitch
route2act: 3

The random number generator seed is: 17


rowSep: 1.000000

[get_stat_hints]:Found hints in <std.stat> file
[get_stat_hints]:Total cells:17 Number of nets:15

[add_extra_cells]:Added 85 spacer cells to the gate array
total cell length: 144
total block length: 153
block x-span:51  block y-span:130


TIMING FACTOR (COMPUTED) : 4.000000
Using default value of bin.penalty.control:1.000000
Original Average Cell Width:28.000000
Adjusted Average Cell Width:3.000000
numBinsG automatically set to:17
binWidthG = 3
average_cell_width is:3
standard deviation of cell length is:38.629
Could not use resume file: std.res  since it could not be found



THIS IS THE ROUTE COST OF THE ORIGINAL PLACEMENT: 558
ECO pin skipped for net <MCLEAR>
ECO pin skipped for net <MCLEAR>
ECO pin skipped for net <MCLOCK>
ECO pin skipped for net <MCLOCK>
ECO pin skipped for net <B0>
ECO pin skipped for net <B0>
ECO pin skipped for net <B0>
ECO pin skipped for net <$COUNT_2/$FJK3_4$Q>
ECO pin skipped for net <IN1>
ECO pin skipped for net <IN1>

The number of nets with 1 pins is 11
The number of nets with 2 pins is 3
The number of nets with 100 pins or more is 0
Average number of pins per net = 1.214286
The maximum number of pins on a single net is:2

Percentage of Nets Connecting to at least 2 cells:1.00
Percentage of Nets Connecting to at least 3 cells:0.00
Percentage of Nets Connecting to at least 4 cells:0.00
Percentage of Nets Connecting to at least 5 cells:0.00
Percentage of Nets Connecting to at least 6 cells:0.00

bdxlen:87  bdylen:188
l:-19  t:170  r:68  b:-18



THIS IS THE ROUTE COST OF THE CURRENT PLACEMENT: 558



THIS IS THE PENALTY OF THE CURRENT PLACEMENT: 651
ECOs are being incorporated as requested
ECO added cell being processed: <$COUNT_2/$FJK3_4>  ... placed in row:3 x:101
ECO added cell being processed: <$COUNT_2/$FJK3_5>  ... placed in row:3 x:453
Total Wire Length - X-Component:309
Total Wire Length - Y-Component:369
Time Penalty:0
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET      DESIRE
    1                     6                  -45          51
    2                    10                  -41          51
    3                   128                   77          51

LONGEST Block is:3   Its length is:128
total penalty is 163
Removed the cell overlaps --- Will do neighbor interchanges only now

TOTAL INTERCONNECT LENGTH: 678
initialRowControl:   0.000
finalRowControl:   0.000
iter      T      Wire accept Time
TimberWolfSC did NOT call even_the_rows()
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
 the number of net = 15
 the number of tilted segment = 4
 the number of switchable L segment = 0
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET      DESIRE
    1                     8                  -47          55
    2                    14                  -41          55
    3                   134                   79          55

LONGEST Block is:3   Its length is:134
 the number of tilted segment = 4
 the number of switchable L segment = 3
 the number of tilted segment = 4
 the number of switchable L segment = 3
After Feeds are Added:
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8                   -47
  2                 14                   -41
  3                134                    79

LONGEST Row is:3   Its length is:134
Eliminated 0 unused feeds in the longest row
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8 
  2                 14 
  3                134 

LONGEST Row is:3   Its length is:134

Actually added 0 gate array spacers

row lengths after steiner trees:
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET
    1                     8                  -47
    2                    14                  -41
    3                   134                   79

LONGEST Block is:3   Its length is:134
initial total global wire   :	569
initial total timing penalty:	0



Steiner cell swap and rotation optimization
swap_limit:0
reduction:	-8		total_red:-8
iterations              :	4
final total global wire :	561
final total time penalty:	0

Total global wire reduced by:	1.406%

VERIFICATION
final total global wire :	561
final total time penalty:	0




THIS IS THE ORIGINAL NUMBER OF TRACKS: 10


no. of accepted flips: 108
no. of attempted flips: 375
THIS IS THE NUMBER OF TRACKS: 10


THIS G. ROUTING IS BEING SAVED AS BEST SO FAR

FINAL NUMBER OF ROUTING TRACKS: 10

MAX OF CHANNEL:  1  is:   2
MAX OF CHANNEL:  2  is:   1
MAX OF CHANNEL:  3  is:   2
MAX OF CHANNEL:  4  is:   5
Confirming number of eliminated feeds:0
Actual # of Feed Cells Added:	6


TimberWolfSC did NOT call even_the_rows()
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
 the number of net = 15
 the number of tilted segment = 4
 the number of switchable L segment = 0
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET      DESIRE
    1                     8                  -47          55
    2                    14                  -41          55
    3                   134                   79          55

LONGEST Block is:3   Its length is:134
 the number of tilted segment = 4
 the number of switchable L segment = 3
 the number of tilted segment = 4
 the number of switchable L segment = 3
After Feeds are Added:
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8                   -47
  2                 14                   -41
  3                134                    79

LONGEST Row is:3   Its length is:134
Eliminated 0 unused feeds in the longest row
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8 
  2                 14 
  3                134 

LONGEST Row is:3   Its length is:134

Actually added 0 gate array spacers

row lengths after steiner trees:
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET
    1                     8                  -47
    2                    14                  -41
    3                   134                   79

LONGEST Block is:3   Its length is:134
initial total global wire   :	569
initial total timing penalty:	0



Steiner cell swap and rotation optimization
swap_limit:0
reduction:	-6		total_red:-6
iterations              :	4
final total global wire :	563
final total time penalty:	0

Total global wire reduced by:	1.054%

VERIFICATION
final total global wire :	563
final total time penalty:	0




THIS IS THE ORIGINAL NUMBER OF TRACKS: 11


no. of accepted flips: 0
no. of attempted flips: 375
THIS IS THE NUMBER OF TRACKS: 11


Actual # of Feed Cells Added:	6


TimberWolfSC did NOT call even_the_rows()
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
 the number of net = 15
 the number of tilted segment = 4
 the number of switchable L segment = 0
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET      DESIRE
    1                     8                  -47          55
    2                    14                  -41          55
    3                   134                   79          55

LONGEST Block is:3   Its length is:134
 the number of tilted segment = 4
 the number of switchable L segment = 3
 the number of tilted segment = 4
 the number of switchable L segment = 3
After Feeds are Added:
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8                   -47
  2                 14                   -41
  3                134                    79

LONGEST Row is:3   Its length is:134
Eliminated 0 unused feeds in the longest row
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8 
  2                 14 
  3                134 

LONGEST Row is:3   Its length is:134

Actually added 0 gate array spacers

row lengths after steiner trees:
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET
    1                     8                  -47
    2                    14                  -41
    3                   134                   79

LONGEST Block is:3   Its length is:134
initial total global wire   :	575
initial total timing penalty:	0



Steiner cell swap and rotation optimization
swap_limit:0
reduction:	-18		total_red:-18
iterations              :	4
final total global wire :	557
final total time penalty:	0

Total global wire reduced by:	3.130%

VERIFICATION
final total global wire :	557
final total time penalty:	0




THIS IS THE ORIGINAL NUMBER OF TRACKS: 10


no. of accepted flips: 1
no. of attempted flips: 375
THIS IS THE NUMBER OF TRACKS: 9


THIS G. ROUTING IS BEING SAVED AS BEST SO FAR

FINAL NUMBER OF ROUTING TRACKS: 9

MAX OF CHANNEL:  1  is:   2
MAX OF CHANNEL:  2  is:   1
MAX OF CHANNEL:  3  is:   3
MAX OF CHANNEL:  4  is:   3
Confirming number of eliminated feeds:0
Actual # of Feed Cells Added:	6


TimberWolfSC did NOT call even_the_rows()
TimberWolfSC using absolute_minimum_feeds
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
 the number of net = 15
 the number of tilted segment = 4
 the number of switchable L segment = 0
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET      DESIRE
    1                     8                  -47          55
    2                    14                  -41          55
    3                   134                   79          55

LONGEST Block is:3   Its length is:134
 the number of tilted segment = 4
 the number of switchable L segment = 3
 the number of tilted segment = 4
 the number of switchable L segment = 3
After Feeds are Added:
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8                   -47
  2                 14                   -41
  3                134                    79

LONGEST Row is:3   Its length is:134
Eliminated 0 unused feeds in the longest row
BLOCK      TOTAL CELL LENGTHS      OVER/UNDER TARGET
  1                  8 
  2                 14 
  3                134 

LONGEST Row is:3   Its length is:134

Actually added 0 gate array spacers

row lengths after steiner trees:
BLOCK    TOTAL CELL LENGTHS    OVER/UNDER TARGET
    1                     8                  -47
    2                    14                  -41
    3                   134                   79

LONGEST Block is:3   Its length is:134
initial total global wire   :	569
initial total timing penalty:	0



Steiner cell swap and rotation optimization
swap_limit:0
reduction:	-6		total_red:-6
iterations              :	4
final total global wire :	563
final total time penalty:	0

Total global wire reduced by:	1.054%

VERIFICATION
final total global wire :	563
final total time penalty:	0




THIS IS THE ORIGINAL NUMBER OF TRACKS: 13


no. of accepted flips: 2
no. of attempted flips: 375
THIS IS THE NUMBER OF TRACKS: 11


Actual # of Feed Cells Added:	6


TimberWolfSC did call even_the_rows()
Total Wire Length - X-Component:392
Total Wire Length - Y-Component:369
Time Penalty:0
